.github/workflows/ci.yaml

```diff
@@ -0,0 +1,30 @@
+name: CI build
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Build & Test
+        run: |
```

Comment:
Worth encapsulating in a makefile?

Reply:
yeah will do if these steps grow and we need to add more commands / flags.